src/Microsoft.AspNet.OData.Shared/Formatter/IODataModelBinderSettings.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.OData.Formatter
+{
+    /// <summary>
+    /// Settings to use during model binding.
+    /// </summary>
+    public interface IODataModelBinderSettings
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether request body binding should be case insensitive.
+        /// </summary>
+        /// <value>The default value is <c>false</c>.</value>
+        bool EnableCaseInsensitiveModelBinding { get; set; }
```

Comment:
why don't use "ODataOptions" to enable it configuration?

Reply:
We are already injecting `ODataOptions` in `AddOData` method of `ODataServiceCollectionExtensions`.
`services.AddSingleton<ODataOptions>();`
I didn't find a clean way of  injecting `ODataOptions` with `EnableCaseInsensitiveModelBinding` configuration.

Reply:
Emm, in this way, can we add a overload to "AddOData" to accept the bool value.

Reply:
To inject ODataOptions is to register ODataOptions again after call "AddOData". you can try it

Reply:
I agree; i'd rather not invent something new here for case insensitivity, but make it part of ODataOptions (even if you have to go through slightly more work to set it).

---
In reply to: [470869995](https://github.com/OData/WebApi/pull/2249#discussion_r470869995) [](ancestors = 470869995)

Reply:
It can inject in the Config(...) before the MapODataService?

---
In reply to: [484275440](https://github.com/OData/WebApi/pull/2249#discussion_r484275440) [](ancestors = 484275440)

Reply:
@xuzhg @mikepizzo  I have updated the PR to use `ODataOptions`

Reply:
#Resolved